syntax = "proto3";
package grade;

import "instructions.proto";

message AICriterionGradingStarted {
    string version_uuid = 1;
    string criterion_uuid = 2;
    string submission_version_uuid = 3;
    instructions.InstructionInfo instruction_info = 4;
    uint64 time_started = 5;
}

message GradingMethod {
    string uuid = 1;
    string model_name = 2;
    string method_name = 3;
    string git_hash = 4;
}

message Grade {
    optional float score = 1;
    string assessment = 2;
}

message AICriterionGrade {
    string grading_started_version_uuid = 1;
    string grading_method_uuid = 4;
    Grade grade = 5;
    uint64 time_finished = 6;
}

message AIInferredCriterionGrade {
    string grading_started_version_uuid = 1;
    string grading_method_uuid = 4;
    string previous_criterion_grade_version_uuid = 5;
    string faculty_override_criterion_grade_version_uuid = 6;
    Grade grade = 7;
    uint64 time_finished = 8;
}

message InstructorCriterionGrade {
    string version_uuid = 1;
    string criterion_uuid = 2;
    string submission_version_uuid = 3;
    string instructor_public_uuid = 4;
    Grade grade = 5;
}

message InstructorOverrideCriterionGrade {
    string version_uuid = 1;
    string criterion_uuid = 2;
    string submission_version_uuid = 3;
    string previous_criterion_grade_version_uuid = 4;
    string instructor_public_uuid = 5;
    Grade grade = 6;
}

message CriterionGradeEmbedding {
    string version_uuid = 1;
    string criterion_grade_version_uuid = 2;
    string embedder_uuid = 3;
    repeated float embedding = 4;
}

message InstructorSubmissionGradeApproval {
    string version_uuid = 1;
    string submission_version_uuid = 2;
    string instructor_public_uuid = 3;
    repeated string criterion_grade_version_uuids = 4;
    uint64 version_timestamp = 5;
}

